### ISSUE TITLE
sphere_eclipse, blink and blink are not differentiable.

### ISSUE THREAD

**Issue by StuartLittlefair** [StuartLittlefair/jax_roche#1]
Consider the following minimum working example that demonstrates the issue:

The MWE is designed to replicate a sphere of radius 0.4, and a vector pointing to earth of (1, 0, 0) fails to differentiate. Grads are all zero. MWE below

```python
    def inner_mwe(phi, y):
        rsphere = 0.4
        yproj = jnp.tan(2.0*jnp.pi*phi) + y
        return jnp.where(yproj <= rsphere, 1.0, 0.0)

    def f_mwe(phi):
        y = jnp.linspace(0.39, 0.41, 5000)
        # still doesn't work with alternative approach below
        #return jnp.sum(jnp.array([inner_mwe(phi, yv) for yv in y]))
        vmapped_inner = vmap(inner_mwe, in_axes=(None, 0))
        return jnp.sum(vmapped_inner(phi, y))
```

If we try and differentiate this, then `grad(f_mwe)(0.0)` is 0.

The problem is the discontinuity in inner_mwe. It's not differentiable! Blink, FBlink and similar functions have the same issue. If we use, e.g a sigmoid function to make the transition from visible to not visible continuous, we recover the gradient:

```python
    def inner_mwe(phi, y):
        rsphere = 0.4
        yproj = jnp.tan(2.0*jnp.pi*phi) + y
        x = (yproj-rsphere)/0.001
        return jax.nn.sigmoid(x)
```

```python
In [314]: f_mwe(0.0) # should be 2500
Out[314]: Array(2499.9976, dtype=float32)

In [315]: delta = 0.000001

In [316]: (f_mwe(delta)-f_mwe(0.0))/delta
Out[316]: Array(1571289.1, dtype=float32)

In [317]: grad(f_mwe)(0.0)
Out[317]: Array(1570338.5, dtype=float32, weak_type=True)
```
